VAMobile/src/screens/HomeScreen/ProfileScreen/SettingsScreen/DeveloperScreen/OverrideApiScreen.tsx

```diff
@@ -0,0 +1,542 @@
+      },
+    ],
+  },
+  {
+    name: 'Direct Deposit',
+    endpoints: [
+      {
+        endpoint: '/v0/payment-information/benefits',
+      },
+    ],
+  },
+  {
+    name: 'Disability Rating',
+    endpoints: [
+      {
+        endpoint: '/v0/disability-rating',
+      },
+    ],
+  },
+  {
+    name: 'Facilities',
+    endpoints: [
+      {
+        endpoint: '/v0/facilities-info',
+      },
+    ],
+  },
+  {
+    name: 'Letters',
+    endpoints: [
+      {
+        endpoint: '/v0/letters/beneficiary',
+      },
+      {
+        endpoint: '/v0/letters',
+      },
+    ],
+  },
+  {
+    name: 'Military Service',
+    endpoints: [
+      {
+        endpoint: '/v0/military-service-history',
+      },
+    ],
+  },
+  {
+    name: 'Notifications',
+    endpoints: [
+      {
+        endpoint: '/v0/push/prefs/',
+      },
+    ],
+  },
+  {
+    name: 'Payments',
+    endpoints: [
+      {
+        endpoint: '/v0/payment-history',
+      },
+    ],
+  },
+  {
+    name: 'Personal Information',
+    endpoints: [
+      {
+        endpoint: '/v2/user',
+      },
+    ],
+  },
+  {
+    name: 'Prescriptions',
+    endpoints: [
+      {
+        endpoint: '/v0/health/rx/prescriptions',
+      },
+      {
+        endpoint: '/tracking',
+      },
+    ],
+  },
+  {
+    name: 'Secure Messaging',
+    endpoints: [
+      {
+        endpoint: '/v0/messaging/health/folders/${folderID}/messages',
+      },
+      {
+        endpoint: '/v0/messaging/health/folders',
+      },
+      {
+        endpoint: '/v0/messaging/health/messages/',
+      },
+      {
+        endpoint: '/v0/messaging/health/messages/recipients',
+      },
+      {
+        endpoint: '/v0/messaging/health/messages/signature',
+      },
+      {
+        endpoint: '/thread',
+      },
+    ],
+  },
+  {
+    name: 'Vaccines',
+    endpoints: [
+      {
+        endpoint: '/v1/health/immunizations',
+      },
+      {
+        endpoint: '/v0/health/locations/',
+      },
+    ],
+  },
+]
+
+const ApiGroupingDisplay = (
+  apiGroupings: {
+    name: string
+    endpoints: {
+      endpoint: string
+    }[]
+  }[],
+  overrideErrors: APIError[],
+  setErrors: React.Dispatch<React.SetStateAction<APIError[]>>,
+  clearErrors: boolean,
+) => {
+  const theme = useTheme()
+  const groupings = apiGroupings.map((group) => {
+    const individualQueries = group.endpoints.map((endpoint, idx) => {
+      return (
+        <Box
+          mt={theme.dimensions.standardMarginBetween}
+          mb={idx === group.endpoints.length - 1 ? theme.dimensions.standardMarginBetween : undefined}>
+          {IndividualQueryDisplay(endpoint.endpoint, overrideErrors, setErrors, clearErrors)}
+        </Box>
+      )
+    })
+    return (
+      <TextArea>
+        <TextView accessibilityRole="header" variant="MobileBodyBold">
+          {group.name}
+        </TextView>
+        {individualQueries}
+      </TextArea>
+    )
+  })
+  return <Box>{groupings}</Box>
+}
```

Comment:
Since this is only used in one place, and it's not much code, can this be moved directly into the screen component? A lot of the props for this component are being passed from the screen component anyway

Reply:
good call out